src/azure-cli-core/azure/cli/core/auth/credential_adaptor.py

```diff
@@ -41,3 +50,33 @@ def get_auxiliary_tokens(self, *scopes, **kwargs):
             return [build_sdk_access_token(cred.acquire_token(list(scopes), **kwargs))
                     for cred in self._auxiliary_credentials]
         return None
+
+
+def _prepare_msal_kwargs(options=None):
+    # Preserve supported options and discard unsupported options (tenant_id, enable_cae).
+    # Both get_token's kwargs and get_token_info's options are accepted as their schema is the same (at least for now).
+    msal_kwargs = {}
+    if options:
+        # For VM SSH. 'data' support is a CLI-specific extension.
+        # SDK doesn't support 'data': https://github.com/Azure/azure-sdk-for-python/pull/16397
+        if 'data' in options:
+            msal_kwargs['data'] = options['data']
+        # For CAE
+        if 'claims' in options:
+            msal_kwargs['claims_challenge'] = options['claims']
+    return msal_kwargs
+
+
+def _build_sdk_access_token_info(token_entry):
```

Comment:
`build_sdk_access_token` is still used by https://github.com/Azure/azure-cli/blob/a2877866d971b685f87462460b735b64dca7be1d/src/azure-cli-core/azure/cli/core/auth/adal_authentication.py#L36

so it can't be moved to `src/azure-cli-core/azure/cli/core/auth/credential_adaptor.py` right now.